pkg/kubectl/cmd/util/factory.go

```diff
@@ -354,6 +354,19 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 					return nil, errors.New("provided options object is not a PodLogOptions")
 				}
 				return c.Pods(t.Namespace).GetLogs(t.Name, opts), nil
+
+			case *api.ReplicationController:
+				opts, ok := options.(*api.PodLogOptions)
+				if !ok {
+					return nil, errors.New("provided options object is not a PodLogOptions")
+				}
+				selector := labels.SelectorFromSet(t.Spec.Selector)
+				pod, err := GetFirstPod(c, t.Namespace, selector)
```

Comment:
I'm also using the "get first pod" logic for a new "debug" command that
creates an exact clone of a pod or object with pod template and starts a
shell in it (with various tweaks for security / environment settings), so
that users who are crash looping have a mechanism to debug.  In that case
the "first pod is good enough" argument applies, because you want to reduce
the friction point for getting context.  The exact pod doesn't matter that
much, as long as the one you pick is stable (as Jeff mentions)

On Tue, Feb 16, 2016 at 6:46 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/util/factory.go
> <https://github.com/kubernetes/kubernetes/pull/21335#discussion_r53099904>
> :
>
> > @@ -354,6 +354,19 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
> >  					return nil, errors.New("provided options object is not a PodLogOptions")
> >  				}
> >  				return c.Pods(t.Namespace).GetLogs(t.Name, opts), nil
> > +
> > +			case *api.ReplicationController:
> > +				opts, ok := options.(*api.PodLogOptions)
> > +				if !ok {
> > +					return nil, errors.New("provided options object is not a PodLogOptions")
> > +				}
> > +				selector := labels.SelectorFromSet(t.Spec.Selector)
> > +				pod, err := GetFirstPod(c, t.Namespace, selector)
>
> Some people have asked for a way to get all logs, also. #19873
> <https://github.com/kubernetes/kubernetes/issues/19873>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/21335/files#r53099904>.
>